metricbeat/module/haproxy/haproxy.go

```diff
@@ -0,0 +1,244 @@
+	ConnRateLimit              string
+	MaxConnRate                string
+	SessRate                   string
+	SessRateLimit              string
+	MaxSessRate                string
+	SslRate                    string
+	SslRateLimit               string
+	MaxSslRate                 string
+	SslFrontendKeyRate         string
+	SslFrontendMaxKeyRate      string
+	SslFrontendSessionReusePct string
+	SslBackendKeyRate          string
+	SslBackendMaxKeyRate       string
+	SslCacheLookups            string
+	SslCacheMisses             string
+	CompressBpsIn              string
+	CompressBpsOut             string
+	CompressBpsRateLim         string
+	ZlibMemUsage               string
+	MaxZlibMemUsage            string
+	Tasks                      string
+	RunQueue                   string
+	IdlePct                    string
+	Node                       string
+	Description                string
+}
+
+// Client is an instance of the HAProxy client
+type Client struct {
+	connection  net.Conn
+	Address     string
+	ProtoScheme string
+}
+
+// NewHaproxyClient returns a new instance of HaproxyClient
+func NewHaproxyClient(address string) (*Client, error) {
+	parts := strings.Split(address, "://")
+	if len(parts) != 2 {
+		return nil, errors.New("Must have protocol scheme and address!")
+	}
+
+	if parts[0] != "tcp" && parts[0] != "unix" {
+		return nil, errors.New("Invalid Protocol Scheme!")
+	}
+
+	return &Client{
+		Address:     parts[1],
+		ProtoScheme: parts[0],
+	}, nil
+}
+
+// Run sends a designated command to the haproxy stats socket
+func (c *Client) run(cmd string) (*bytes.Buffer, error) {
+	var conn net.Conn
+	response := bytes.NewBuffer(nil)
+
+	conn, err := net.Dial(c.ProtoScheme, c.Address)
+	if err != nil {
+		return response, err
+	}
+	c.connection = conn
+
+	defer c.connection.Close()
+
+	_, err = c.connection.Write([]byte(cmd + "\n"))
+	if err != nil {
+		return response, err
+	}
+
+	_, err = io.Copy(response, c.connection)
+	if err != nil {
+		return response, err
+	}
+
+	if strings.HasPrefix(response.String(), "Unknown command") {
+		return response, fmt.Errorf("Unknown command: %s", cmd)
+	}
+
+	return response, nil
+}
+
+// GetStat returns the result from the 'show stat' command
+func (c *Client) GetStat() (statRes []*Stat, err error) {
+
+	runResult, err := c.run("show stat")
+	if err != nil {
+		return nil, err
+	}
+
+	csvReader := csv.NewReader(runResult)
+	csvReader.TrailingComma = true
+
+	err = gocsv.UnmarshalCSV(csvReader, &statRes)
+	if err != nil {
+		return nil, fmt.Errorf("Error parsing CSV: %s", err)
+	}
+
+	return statRes, nil
+
+}
+
+// GetInfo returns the result from the 'show stat' command
+func (c *Client) GetInfo() (infoRes *Info, err error) {
+
+	res, err := c.run("show info")
+	if err != nil {
+		return nil, err
+	}
+
+	if b, err := ioutil.ReadAll(res); err == nil {
+
+		resultMap := map[string]interface{}{}
+
+		for _, ln := range strings.Split(string(b), "\n") {
+
+			ln := strings.Trim(ln, " ")
```

Comment:
There is the [`strings.TrimSpace`](https://golang.org/pkg/strings/#TrimSpace) function for this.